docs/ContainerRuntimeConfigDesign.md

```diff
@@ -0,0 +1,95 @@
+# Summary
+
+Users need a way to update the container runtime configuration. Users can do this via MCO currently, but they need to know the correct options to use. We can hopefully help customers by documenting a CRD with the ContainerRuntime tuneables and have the MCO use these tuneables when rendering the crio.conf, storage.conf, and registries.conf files to Ignition/disk.
+
+The CRI-O runtime is used by default and is the recommended solution.
+
+# Motivation
+
+The MCO contains the logic to write the crio.conf, storage.conf, and registries.conf configuration files to Ignition. When Ignition starts on a machine it writes these three files to configure the container runtime. When these resources change within the MachineConfig, the new configs are rewritten and the MachineConfigDaemon is instructed to reboot the machine to pick up the new configs.
+
+## Goals
+
+The following goals should not require explicit knowledge of a customer to know the right command-line or configuration options to use within the ContainerRuntime.
+
+1. setting pids limit
+
+2. setting log level
+
+3. configuring list of search registries
+
+4. configuring list of insecure registries
+
+5. configuring list of blocked registries
+
+6. configuring max size of storage driver
```

Comment:
We need to add infra_image to this list.

Reply:
This seems like it is more per machine set, is that correct?

Reply:
I think configuring overlay is distinct from machine api.  We also need this in UPI modes where machines are not necessarily present.